src/core/mac/sub_mac_csl_receiver.cpp

```diff
@@ -54,6 +54,24 @@ void SubMac::CslInit(void)
     mCslTimer.Stop();
 }
 
+void SubMac::RestartCslTimerAfterSyncUpdate(void)
+{
```

Comment:
I suggest moving the check of `RadioSupportsReceiveTiming` inside the function. And add a comment for the function that this only applies for the case where radio supports receive timing.